src/main/scala/com/qubole/spark/datasources/hiveacid/sql/catalyst/plans/command/MergeCommand.scala

```diff
@@ -46,12 +48,32 @@ case class MergeCommand(targetTable: LogicalPlan,
       case _ => throw HiveAcidErrors.mergeValidationError("WHEN NOT Clause has to be INSERT CLAUSE")
     }
 
-    children.head match {
-      case LogicalRelation(relation: HiveAcidRelation, _, _ , _) =>
-        relation.merge(SqlUtils.logicalPlanToDataFrame(sparkSession, sourceTable),
+    val targetRelation = children.head
+    val sourceRelation = children.last
+    val sourceTableFullyQualifiedName = EliminateSubqueryAliases(sourceRelation) match {
```

Comment:
this can have side effects. for current scenario its fine in future if we extend merge to generic plan this will be bad. Can you use the normal case match here ?

Reply:
Done